### DISCUSSION TITLE
How to tell if a stream isn't running

### DISCUSSION THREAD

**Discussion by 0xAl3xH** [bluenviron/mediamtx#2963] [Questions]
### Question

I'm wondering what the best way is to tell whether a stream is running on a URL or not. I see that a `GET` request to the endpoint returns `404` in my testing but I'm not sure if this is sufficient or if there's another more official way to query for this status.

**Comment by aler9** [comment]
Hello, in my opinion if you want to get an overall view of what is running or not on the server, you can use the API and in particular the `/paths/list` endpoint:

```
curl http://127.0.0.1:9997/v3/paths/list
```

resulting in

```json
{
  "pageCount": 0,
  "items": [
    {
      "name": "string",
      "confName": "string",
      "source": {
        "type": "hlsSource",
        "id": "string"
      },
      "ready": true,
      "readyTime": "string",
      "tracks": [
        "string"
      ],
      "bytesReceived": 0,
      "bytesSent": 0,
      "readers": [
        {
          "type": "hlsMuxer",
          "id": "string"
        }
      ]
    }
  ]
}
```

As you can see there's a `ready` field that tells whether a stream is available or not.

[a comment by an automated account is not shown]
